pkg/controller/component/vars.go

```diff
@@ -381,15 +381,15 @@ func resolveClusterObjectRefVars(ctx context.Context, cli client.Reader, synthes
 	vars1, vars2 := make([]corev1.EnvVar, 0), make([]corev1.EnvVar, 0)
 	for _, v := range definedVars {
 		switch {
-		case len(v.Value) > 0:
-			vars1 = append(vars1, corev1.EnvVar{Name: v.Name, Value: v.Value})
 		case v.ValueFrom != nil:
-			var1, var2, err := resolveClusterObjectVarRef(ctx, cli, synthesizedComp, v.Name, *v.ValueFrom)
+			var1, var2, err := resolveClusterObjectVarRef(ctx, cli, synthesizedComp, v.Name, *v.ValueFrom, v)
 			if err != nil {
 				return nil, nil, err
 			}
 			vars1 = append(vars1, var1...)
 			vars2 = append(vars2, var2...)
+		case len(v.Value) > 0:
+			vars1 = append(vars1, corev1.EnvVar{Name: v.Name, Value: v.Value})
```

Comment:
Is there any significance to the order of "case v.ValueFrom != nil" and "case len(v.Value) > 0"?

Reply:
This order is introduced by this pr to ensure that ValueFrom can be processed before Value, so as to use Value as the default value.